app/models/concerns/document_attachment.rb

```diff
@@ -1,24 +1,13 @@
 module DocumentAttachment
   extend ActiveSupport::Concern
+  include S3Headers
```

Comment:
nice tidy up! 

i wonder if you can use a kind of of `acts_as` like pattern
https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/blob/master/app/services/simple_bill_typeable.rb
but might not work within a containing `concern`